packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -328,35 +329,26 @@ class AssetImage extends AssetBundleImageProvider {
     return completer.future;
   }
 
-  /// Parses the asset manifest string into a strongly-typed map.
+  /// Decodes the asset manifest's file contents into it's Dart representation.
   @visibleForTesting
-  static Future<Map<String, List<String>>?> manifestParser(String? jsonData) {
-    if (jsonData == null) {
-      return SynchronousFuture<Map<String, List<String>>?>(null);
-    }
-    // TODO(ianh): JSON decoding really shouldn't be on the main thread.
-    final Map<String, dynamic> parsedJson = json.decode(jsonData) as Map<String, dynamic>;
-    final Iterable<String> keys = parsedJson.keys;
-    final Map<String, List<String>> parsedManifest = <String, List<String>> {
-      for (final String key in keys) key: List<String>.from(parsedJson[key] as List<dynamic>),
-    };
-    // TODO(ianh): convert that data structure to the right types.
-    return SynchronousFuture<Map<String, List<String>>?>(parsedManifest);
+  static Map<dynamic, dynamic> decodeAssetManifest(ByteData data) {
+    const StandardMessageCodec codec = StandardMessageCodec();
+    return codec.decodeMessage(data) as Map<dynamic, dynamic>;
   }
 
-  String? _chooseVariant(String main, ImageConfiguration config, List<String>? candidates) {
+  dynamic _chooseVariant(String main, ImageConfiguration config, List<dynamic>? candidates) {
```

Comment:
_chooseVariant should return the string that is needed to load the asset in question. Any additional facets used to select this asset should be passed into this function (i.e. brightness)